middleware-catalog-module/address/Address/src/main/java/co/tpg/catalog/function/AddressFunction.java

```diff
@@ -0,0 +1,156 @@
+import co.tpg.catalog.response.ErrorResponse;
+import co.tpg.catalog.response.AddressListResponse;
+import co.tpg.catalog.response.AddressResponse;
+import co.tpg.catalog.dao.AddressDAO;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
+
+/**
+ * Handler for requests to Address.
+ * @author maxx
+ * @since 2019-10-11
+ */
+public class AddressFunction implements RequestHandler<AddressRequest, AbstractResponse> {
+
+    public AbstractResponse handleRequest(final AddressRequest input, final Context context) {
+        final String DYNAMO_TABLE_NAME = "Address";
+        final AmazonDynamoDB dynamoDB = AmazonDynamoDBClientBuilder.defaultClient();
+        final AddressResponse response = new AddressResponse();
+        final AddressDAO addressDAO = new AddressDAO();
+        final ErrorResponse errorResponse = new ErrorResponse();
+        final Map<String, String> headers = new HashMap<>();
+        final DynamoDBMapper mapper = new DynamoDBMapper(dynamoDB);
+        String id;
+
+        headers.put("Content-Type", "application/json");
+        response.setHeaders(headers);
+
+        errorResponse.setStatusCode(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        context.getLogger().log(String.format("HTTP method: %s\n",input.getHttpMethod()));
+
+        try {
+            switch (input.getHttpMethod()) {
+                case HttpMethod.POST: {
+                    try {
+                        Address address = input.getBody();
+                        address.setId(UUID.randomUUID().toString());
+                        context.getLogger().log(String.format("POST body=%s\n", input.getBody()));
+                        mapper.save(address);
+                        response.setBody(address);
+                        response.setStatusCode(HttpServletResponse.SC_CREATED);
+                    } catch (ResourceNotFoundException ex) {
+                        errorResponse.setBody(ProcessingException.builder().message(String.format("The table named %s could not be found in the backend system.", DYNAMO_TABLE_NAME)).build());
+                        return errorResponse;
+                    } catch (AmazonServiceException ex) {
+                        errorResponse.setBody(ProcessingException.builder().message(ex.getMessage()).build());
+                        return errorResponse;
+                    }
+                    break;
+                }
+                case HttpMethod.GET:
```

Comment:
Same for GET and other methods. See in the master branch, the class TeachingClassFunction.java. It is using a DAO (Data Access Object design pattern) class to access data. See here: https://github.com/tpg-nz/schoolncloud/blob/master/middleware-catalog-module/teaching-class/TeachingClass/src/main/java/co/tpg/catalog/function/TeachingClassFunction.java. It is using a more cohesive approach and it is reducing the code size in the lambda function.